### PR TITLE
Add field names to pallet `Event` variants

### PR DESCRIPTION
**Pull request by ayevbeosa**
This pull request changes the pallet `Events` variants from tuples to `structs` with named fields.

Part implementation for [#9903](https://github.com/paritytech/substrate/issues/9903) and [#9993](https://github.com/paritytech/substrate/pull/9993).

### Pallets With Changes

- uniques
- utility
- vesting

### CODE REVIEW

File: utils/frame/remote-externalities/Cargo.toml [hunk 1]
```diff
@@ -29,6 +29,8 @@ sp-core = { version = "4.0.0-dev", path = "../../../primitives/core" }
 sp-runtime = { version = "4.0.0-dev", path = "../../../primitives/runtime" }
 sp-version = { version = "4.0.0-dev", path = "../../../primitives/version" }
 
+frame-support = { version = "4.0.0-dev", path = "../../../frame/support" }
```
Comment (athei) [inline utils/frame/remote-externalities/Cargo.toml:32]: Why?

File: frame/uniques/src/lib.rs [hunk 2]
```diff
@@ -192,62 +192,95 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config<I>, I: 'static = ()> {
 		/// An asset class was created. \[ class, creator, owner \]
```
Comment (athei) [inline frame/uniques/src/lib.rs:194]: This is now redundant and should be removed whenever we use a struct like variant.

```suggestion
		/// An asset class was created.
```

**Review by athei — CHANGES_REQUESTED** [review]
Comments needs to be updated.

File: frame/uniques/src/lib.rs (hunk 2 above)
Comment (ayevbeosa) [reply frame/uniques/src/lib.rs:194]: Okay thanks. Should I do this change for all other structs too?

File: utils/frame/remote-externalities/Cargo.toml (hunk 1 above)
Comment (ayevbeosa) [reply utils/frame/remote-externalities/Cargo.toml:32]: The compilation kept failing till I added this dependency.

File: frame/uniques/src/lib.rs (hunk 2 above)
Comment (athei) [reply frame/uniques/src/lib.rs]: Yes.

File: utils/frame/remote-externalities/Cargo.toml (hunk 1 above)
Comment (athei) [reply utils/frame/remote-externalities/Cargo.toml:32]: Strange. Wondering why this is. Seems unrelated to this PR. Maybe you need to merge master?

Comment (ayevbeosa) [reply utils/frame/remote-externalities/Cargo.toml:32]: Yeah, it is. I did merge master before sending this PR. Will do so again.

Comment (ayevbeosa) [reply utils/frame/remote-externalities/Cargo.toml:32]: I merged master and removed the dependency and the test failed. Should I leave it then?

Comment (ayevbeosa) [reply utils/frame/remote-externalities/Cargo.toml:32]: image

This is what shows.

Comment (athei) [reply utils/frame/remote-externalities/Cargo.toml:32]: Yeah. My feeling is that something in the CI image was maybe changed. Every pending PR needs to add this line to not fail. Leave it in please move it to `dev-dependencies` as its only used in tests.

Like here: https://github.com/paritytech/substrate/pull/9993/files#diff-fe6fa7aabc85c3fadcb402dc4b2e00618e68ebf6371e80ce3dbfdb4ee9f0b22e

Comment (ayevbeosa) [reply utils/frame/remote-externalities/Cargo.toml:32]: Alright, will do so.

**Review by athei — APPROVED** [review]

File: utils/frame/remote-externalities/Cargo.toml [hunk 3]
```diff
@@ -32,6 +32,7 @@ sp-version = { version = "4.0.0-dev", path = "../../../primitives/version" }
 [dev-dependencies]
 tokio = { version = "1.10", features = ["macros", "rt-multi-thread"] }
 pallet-elections-phragmen = { path = "../../../frame/elections-phragmen", version = "5.0.0-dev" }
+frame-support = { path = "../../../frame/support", version = "4.0.0-dev" }
```
Comment (emostov) [inline utils/frame/remote-externalities/Cargo.toml:35]: What is this for?

**Review by emostov — APPROVED** [review]

File: utils/frame/remote-externalities/Cargo.toml (hunk 3 above)
Comment (emostov) [reply utils/frame/remote-externalities/Cargo.toml:35]: Ah, nvm I see https://github.com/paritytech/substrate/pull/10184#discussion_r743495518

**Comment by athei** [comment]
@ayevbeosa You need to merge master and resolve conflict before we can merge.

**Comment by ayevbeosa** [comment]
> 
> 
> @ayevbeosa You need to merge master and resolve conflict before we can merge.

Doing so now.

**Comment by athei** [comment]
bot merge

**Comment by athei** [comment]
@ayevbeosa You need to provide an address in your opening post to receive a tip. Like [here](https://github.com/paritytech/substrate/pull/10206).

**Comment by ayevbeosa** [comment]
> 
> 
> @ayevbeosa You need to provide an address in your opening post to receive a tip. Like [here](https://github.com/paritytech/substrate/pull/10206).

This has been done. Thanks for the tip.

**Pull request merged**